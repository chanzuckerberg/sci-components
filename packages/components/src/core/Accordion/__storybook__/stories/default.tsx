import { Args } from "@storybook/react";
import RawAccordion, {
  AccordionDetails,
  AccordionHeader,
} from "src/core/Accordion";
import { EXTRA_SHORT_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Accordion = (props: Args): JSX.Element => {
  const { id, subtitle, useDivider, togglePosition } = props;
  return (
    <RawAccordion
      id={id}
      useDivider={useDivider}
      togglePosition={togglePosition}
      {...props}
    >
      <AccordionHeader id={`${id}-header`} subtitle={subtitle}>
        Accordion Header
      </AccordionHeader>
      <AccordionDetails>{EXTRA_SHORT_LOREM_IPSUM}</AccordionDetails>
    </RawAccordion>
  );
};
