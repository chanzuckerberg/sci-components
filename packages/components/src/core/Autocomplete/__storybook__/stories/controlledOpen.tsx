import { Args } from "@storybook/types";
import { DefaultAutocompleteOption } from "src/core/Autocomplete/components/AutocompleteBase";
import { AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS } from "src/common/storybook/AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS";
import { useState } from "react";
import RawAutocomplete from "src/core/Autocomplete";
import { AUTOCOMPLETE_WRAPPER_STYLES } from "../constants";

export const ControlledOpenDemo = <T extends DefaultAutocompleteOption>(
  props: Args
): JSX.Element => {
  const {
    label,
    multiple,
    options = AUTOCOMPLETE_SINGLE_COLUMN_OPTIONS,
  } = props;

  const [open, setOpen] = useState(true);

  return (
    <div style={AUTOCOMPLETE_WRAPPER_STYLES}>
      <RawAutocomplete
        onClickAway={() => {
          setOpen(false);
        }}
        onClick={() => {
          if (open) {
            setOpen(false);
          } else {
            setOpen(true);
          }
        }}
        open={open}
        id="autocomplete-demo"
        disableCloseOnSelect={multiple}
        label={label}
        multiple={multiple}
        options={options}
        getOptionDisabled={(option: T) => {
          return option.name === "Type: feature request";
        }}
        {...props}
      />
    </div>
  );
};
