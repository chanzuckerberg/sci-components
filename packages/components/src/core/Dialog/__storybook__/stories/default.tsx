import { Args } from "@storybook/react";
import { useState } from "react";
import Button from "src/core/Button";
import RawDialog from "src/core/Dialog";
import DialogTitle from "src/core/Dialog/components/DialogTitle";
import DialogContent from "src/core/Dialog/components/DialogContent";
import DialogActions from "src/core/Dialog/components/DialogActions";
import { EXTRA_LONG_LOREM_IPSUM } from "src/common/storybook/loremIpsum";

export const Dialog = (props: Args): JSX.Element => {
  const {
    buttonPosition,
    longContent = false,
    isOpen: propsIsOpen = false,
    titleOnClose,
    ...rest
  } = props;

  const [isOpen, setIsOpen] = useState(propsIsOpen);

  return (
    <>
      <Button sdsStyle="minimal" sdsType="primary" onClick={handleClick}>
        Open {rest.sdsSize}
      </Button>
      <RawDialog onClose={handleClose} open={isOpen} {...rest}>
        <DialogTitle
          title="Title"
          subtitle="Optional Subtitle"
          overline="Optional Overline"
          onClose={titleOnClose ? handleClose : undefined}
          data-testid="dialog-title"
        />
        <DialogContent data-testid="dialog-content">
          {longContent ? EXTRA_LONG_LOREM_IPSUM : "Content Module"}
        </DialogContent>
        <DialogActions
          data-testid="dialog-actions"
          buttonPosition={buttonPosition}
        >
          <Buttons />
        </DialogActions>
      </RawDialog>
    </>
  );

  function Buttons(): JSX.Element {
    if (buttonPosition === "left") {
      return (
        <>
          <Button sdsStyle="square" sdsType="primary" onClick={handleClose}>
            Primary Action
          </Button>
          <Button sdsStyle="square" sdsType="secondary" onClick={handleClose}>
            Secondary Action
          </Button>
        </>
      );
    }

    return (
      <>
        <Button sdsStyle="square" sdsType="secondary" onClick={handleClose}>
          Secondary Action
        </Button>
        <Button sdsStyle="square" sdsType="primary" onClick={handleClose}>
          Primary Action
        </Button>
      </>
    );
  }

  function handleClick() {
    setIsOpen(true);
  }

  function handleClose() {
    setIsOpen(false);
  }
};
