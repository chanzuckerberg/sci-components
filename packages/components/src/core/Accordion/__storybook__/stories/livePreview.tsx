import { Args } from "@storybook/react";
import { Accordion } from "./default";

const livePreviewStyles = {
  display: "grid",
  gridColumnGap: "24px",
  gridRowGap: "50px",
  gridTemplateColumns: "repeat(4, 200px)",
};

export function LivePreviewDemo(props: Args): JSX.Element {
  return (
    <div style={livePreviewStyles as React.CSSProperties}>
      <Accordion
        id="accordion-1"
        togglePosition="right"
        useDivider
        {...props}
      />
      <Accordion
        id="accordion-2"
        togglePosition="right"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
      <Accordion id="accordion-3" togglePosition="left" useDivider {...props} />
      <Accordion
        id="accordion-4"
        togglePosition="left"
        subtitle="Optional Subtitle"
        useDivider
        {...props}
      />
    </div>
  );
}
