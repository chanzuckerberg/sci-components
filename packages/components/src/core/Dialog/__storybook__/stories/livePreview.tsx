import { Args } from "@storybook/react";
import { Dialog } from "./default";

export function LivePreviewDemo(props: Args): JSX.Element {
  return (
    <>
      <Dialog sdsSize="xs" {...props} />
      <Dialog sdsSize="s" {...props} />
      <Dialog sdsSize="m" {...props} />
      <Dialog sdsSize="l" {...props} />
    </>
  );
}
