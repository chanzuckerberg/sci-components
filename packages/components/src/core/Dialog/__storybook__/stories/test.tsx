import { Args } from "@storybook/react";
import { Dialog } from "./default";

export function TestDemo(props: Args): JSX.Element {
  const defaultProps = { ...props, isOpen: true, titleOnClose: true };

  return (
    <>
      <Dialog sdsSize="xs" {...defaultProps} />
      <Dialog sdsSize="s" {...defaultProps} />
      <Dialog sdsSize="m" {...defaultProps} />
      <Dialog sdsSize="l" {...defaultProps} />
    </>
  );
}
