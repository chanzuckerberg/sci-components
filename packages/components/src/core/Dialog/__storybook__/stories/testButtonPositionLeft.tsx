import { Args } from "@storybook/react";
import { Dialog } from "./default";

export function TestButtonPositionLeftDemo(props: Args): JSX.Element {
  return <Dialog sdsSize="xs" {...props} isOpen buttonPosition="left" />;
}
