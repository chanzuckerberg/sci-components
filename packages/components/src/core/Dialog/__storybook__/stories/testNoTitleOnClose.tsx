import { Args } from "@storybook/react";
import { Dialog } from "./default";

export function TestNoTitleOnCloseDemo(props: Args): JSX.Element {
  return <Dialog sdsSize="xs" {...props} isOpen />;
}
