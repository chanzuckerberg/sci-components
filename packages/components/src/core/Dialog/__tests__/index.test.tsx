import { generateSnapshots } from "@chanzuckerberg/story-utils";
import { composeStories } from "@storybook/react";
import { render, screen } from "@testing-library/react";
import * as stories from "../__storybook__/index.stories";

const DIALOG_TITLE_TEST_ID = "dialog-title";
const DIALOG_CONTENT_TEST_ID = "dialog-content";
const DIALOG_ACTIONS_TEST_ID = "dialog-actions";

// Returns a component that already contain all decorators from story level, meta level and global level.
const { Test, TestButtonPositionLeft, TestNoTitleOnClose } =
  composeStories(stories);

describe("<Dialog />", () => {
  generateSnapshots(stories);

  test("Dialog all sizes match the snapshots", () => {
    render(<Test />);

    const titles = screen.getAllByTestId(DIALOG_TITLE_TEST_ID);
    const contents = screen.getAllByTestId(DIALOG_CONTENT_TEST_ID);
    const actions = screen.getAllByTestId(DIALOG_ACTIONS_TEST_ID);

    Array.from(Array(titles.length), (_, index) => index).forEach((index) => {
      expect(titles[index]).toMatchSnapshot();
      expect(contents[index]).toMatchSnapshot();
      expect(actions[index]).toMatchSnapshot();
    });
  });

  it("renders Dialog without title X button", () => {
    render(<TestNoTitleOnClose />);

    const title = screen.getByTestId(DIALOG_TITLE_TEST_ID);

    expect(title).toMatchSnapshot();
  });

  it("renders Dialog left positioned buttons", () => {
    render(<TestButtonPositionLeft />);

    const actions = screen.getByTestId(DIALOG_ACTIONS_TEST_ID);

    expect(actions).toMatchSnapshot();
  });
});
