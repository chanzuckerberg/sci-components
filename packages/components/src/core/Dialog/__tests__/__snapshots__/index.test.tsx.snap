// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Default story renders snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-b80hde-MuiButtonBase-root-MuiButton-root"
  data-style="minimal"
  tabindex="0"
  type="button"
>
  Open 
  m
</button>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 1`] = `
<h2
  class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-uq0kkp-MuiTypography-root-MuiDialogTitle-root"
  data-testid="dialog-title"
  id=":r1:"
>
  <button
    aria-label="Close"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-fwkm3g-MuiButtonBase-root-MuiIconButton-root"
    icon="XMark"
    tabindex="0"
    type="button"
  >
    <div
      class="css-1ft6wgl"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-m01104-MuiSvgIcon-root"
        data-file-name="IconXMarkSmall"
        data-testid="IconXMarkSmall"
        fillcontrast="white"
        focusable="false"
        height="16"
        viewBox="0 0 16 16"
        width="16"
      />
    </div>
  </button>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1qnp4ei-MuiTypography-root"
  >
    Optional Overline
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Title
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Optional Subtitle
  </p>
</h2>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 2`] = `
<div
  class="MuiDialogContent-root css-loe64q-MuiDialogContent-root"
  data-testid="dialog-content"
>
  Content Module
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 3`] = `
<div
  class="MuiDialogActions-root MuiDialogActions-spacing css-3tqi0u-MuiDialogActions-root"
  data-testid="dialog-actions"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ke1fod-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Secondary Action
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-p61vfz-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Primary Action
  </button>
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 4`] = `
<h2
  class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-uq0kkp-MuiTypography-root-MuiDialogTitle-root"
  data-testid="dialog-title"
  id=":r2:"
>
  <button
    aria-label="Close"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-slrckh-MuiButtonBase-root-MuiIconButton-root"
    icon="XMark"
    tabindex="0"
    type="button"
  >
    <div
      class="css-1ft6wgl"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-zwwmfn-MuiSvgIcon-root"
        data-file-name="IconXMarkLarge"
        data-testid="IconXMarkLarge"
        fillcontrast="white"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      />
    </div>
  </button>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1qnp4ei-MuiTypography-root"
  >
    Optional Overline
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Title
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Optional Subtitle
  </p>
</h2>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 5`] = `
<div
  class="MuiDialogContent-root css-loe64q-MuiDialogContent-root"
  data-testid="dialog-content"
>
  Content Module
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 6`] = `
<div
  class="MuiDialogActions-root MuiDialogActions-spacing css-3tqi0u-MuiDialogActions-root"
  data-testid="dialog-actions"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ke1fod-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Secondary Action
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-p61vfz-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Primary Action
  </button>
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 7`] = `
<h2
  class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-uq0kkp-MuiTypography-root-MuiDialogTitle-root"
  data-testid="dialog-title"
  id=":r3:"
>
  <button
    aria-label="Close"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-182od9f-MuiButtonBase-root-MuiIconButton-root"
    icon="XMark"
    tabindex="0"
    type="button"
  >
    <div
      class="css-1ft6wgl"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18ki404-MuiSvgIcon-root"
        data-file-name="IconXMarkLarge"
        data-testid="IconXMarkLarge"
        fillcontrast="white"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      />
    </div>
  </button>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1qnp4ei-MuiTypography-root"
  >
    Optional Overline
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Title
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Optional Subtitle
  </p>
</h2>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 8`] = `
<div
  class="MuiDialogContent-root css-loe64q-MuiDialogContent-root"
  data-testid="dialog-content"
>
  Content Module
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 9`] = `
<div
  class="MuiDialogActions-root MuiDialogActions-spacing css-3tqi0u-MuiDialogActions-root"
  data-testid="dialog-actions"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ke1fod-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Secondary Action
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-p61vfz-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Primary Action
  </button>
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 10`] = `
<h2
  class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-uq0kkp-MuiTypography-root-MuiDialogTitle-root"
  data-testid="dialog-title"
  id=":r4:"
>
  <button
    aria-label="Close"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-182od9f-MuiButtonBase-root-MuiIconButton-root"
    icon="XMark"
    tabindex="0"
    type="button"
  >
    <div
      class="css-1ft6wgl"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18ki404-MuiSvgIcon-root"
        data-file-name="IconXMarkLarge"
        data-testid="IconXMarkLarge"
        fillcontrast="white"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      />
    </div>
  </button>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1qnp4ei-MuiTypography-root"
  >
    Optional Overline
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Title
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Optional Subtitle
  </p>
</h2>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 11`] = `
<div
  class="MuiDialogContent-root css-loe64q-MuiDialogContent-root"
  data-testid="dialog-content"
>
  Content Module
</div>
`;

exports[`<Dialog /> Dialog all sizes match the snapshots 12`] = `
<div
  class="MuiDialogActions-root MuiDialogActions-spacing css-3tqi0u-MuiDialogActions-root"
  data-testid="dialog-actions"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ke1fod-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Secondary Action
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-p61vfz-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Primary Action
  </button>
</div>
`;

exports[`<Dialog /> renders Dialog left positioned buttons 1`] = `
<div
  class="MuiDialogActions-root MuiDialogActions-spacing css-1naj6t2-MuiDialogActions-root"
  data-testid="dialog-actions"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-p61vfz-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Primary Action
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-ke1fod-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Secondary Action
  </button>
</div>
`;

exports[`<Dialog /> renders Dialog without title X button 1`] = `
<h2
  class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-uq0kkp-MuiTypography-root-MuiDialogTitle-root"
  data-testid="dialog-title"
  id=":r5:"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1qnp4ei-MuiTypography-root"
  >
    Optional Overline
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Title
  </p>
  <p
    class="MuiTypography-root MuiTypography-body1 css-l7o1a2-MuiTypography-root"
  >
    Optional Subtitle
  </p>
</h2>
`;
